<?xml version="1.0" ?>
<launch>

    <!-- these are the arguments you can pass this launch file, for example paused:=true -->
    <arg name="paused" default="false"/>
    <arg name="use_sim_time" default="true"/>
    <arg name="gui" default="true"/>
    <arg name="headless" default="false"/>
    <arg name="debug" default="false"/>
    <arg name="model" default="$(find swarmus_ros_description)/urdf/pioneer.urdf.xacro"/>

    <!-- define the robots as args -->
    <arg name="robot_1_name" default="agent_1"/>
    <arg name="robot_2_name" default="agent_2"/>
    <arg name="robot_3_name" default="agent_3"/>

    <rosparam param="robot_list" subst_value="True" >[$(arg robot_1_name),$(arg robot_2_name),$(arg robot_3_name))]</rosparam>

    <!-- Launch empty world, HiveConnect (not used) and the world's tf publisher-->
    <include file="$(find swarmus_ros_simulation)/launch/reusable/simulation_services.launch"/>

    <include file="$(find swarmus_ros_simulation)/launch/reusable/pioneer_spawner.launch">
        <arg name="ns" value="$(arg robot_1_name)"/>
        <arg name="robot_name" value="$(arg robot_1_name)"/>
        <arg name="robot_pos_x" value="1.0"/>
        <arg name="yaw" value="1.54"/>
    </include>

    <include file="$(find swarmus_ros_simulation)/launch/reusable/pioneer_spawner.launch">
        <arg name="ns" value="$(arg robot_2_name)"/>
        <arg name="robot_name" value="$(arg robot_2_name)"/>
        <arg name="robot_pos_x" value="2.0"/>
    </include>

    <include file="$(find swarmus_ros_simulation)/launch/reusable/pioneer_spawner.launch">
        <arg name="ns" value="$(arg robot_3_name)"/>
        <arg name="robot_name" value="$(arg robot_3_name)"/>
        <arg name="robot_pos_x" value="7.5"/>
        <arg name="robot_pos_y" value="7.5"/>
        <arg name="yaw" value="3.927"/>
    </include>

    <node pkg="swarmus_ros_simulation" type="communicationBroker" name="communicationBroker" output="screen"/>

</launch>
